ntoskrnl/include/internal/i386/mm.h

```diff
@@ -14,32 +14,27 @@
 /* Memory layout base addresses */
 #define MI_USER_PROBE_ADDRESS                   (PVOID)0x7FFF0000
 #define MI_DEFAULT_SYSTEM_RANGE_START           (PVOID)0x80000000
-#ifndef _X86PAE_
-#define HYPER_SPACE                                    0xC0400000
-#define HYPER_SPACE_END                                0xC07FFFFF
-#else
-#define HYPER_SPACE                                    0xC0800000
-#define HYPER_SPACE_END                                0xC0BFFFFF
-#endif
 #define MI_SYSTEM_CACHE_WS_START                (PVOID)0xC0C00000
 #define MI_SYSTEM_CACHE_START                   (PVOID)0xC1000000
 #define MI_PAGED_POOL_START                     (PVOID)0xE1000000
 #define MI_NONPAGED_POOL_END                    (PVOID)0xFFBE0000
 #define MI_DEBUG_MAPPING                        (PVOID)0xFFBFF000
 #define MI_HIGHEST_SYSTEM_ADDRESS               (PVOID)0xFFFFFFFF
 
-/* Misc address definitions */
+/* Highest virtual address for VAD */
 #define MM_HIGHEST_VAD_ADDRESS \
     (PVOID)((ULONG_PTR)MM_HIGHEST_USER_ADDRESS - (16 * PAGE_SIZE))
-#define MI_MAPPING_RANGE_START              (ULONG)HYPER_SPACE
-#define MI_MAPPING_RANGE_END                (MI_MAPPING_RANGE_START + \
-                                                  MI_HYPERSPACE_PTES * PAGE_SIZE)
-#define MI_DUMMY_PTE                        (PMMPTE)((ULONG_PTR)MI_MAPPING_RANGE_END + \
-                                                  PAGE_SIZE)
-#define MI_VAD_BITMAP                       (PMMPTE)((ULONG_PTR)MI_DUMMY_PTE + \
-                                                  PAGE_SIZE)
-#define MI_WORKING_SET_LIST                 (PMMPTE)((ULONG_PTR)MI_VAD_BITMAP + \
-                                                  PAGE_SIZE)
+
+/* Hyper space definitions */
+#define HYPER_SPACE            (PTE_TOP + 1)
```

Comment:
If I wrote new code
#define MiPteToAddress (_Pte) ((PVOID) ((ULONG) (_ Pte) << 10)),
then they would probably tell me that this is "magic" ...
Now when I answer the question "Why?" (#define MiPteToAddress (_Pte) ((PVOID) ((MiGetPteOffset (_Pte)) * PAGE_SIZE))), you tell me this is an exceptional case. Then the question arises, "Why is this case so exceptional?"

Reply:
I'd rather want to know: What does it fix? Nothing. It's just noise with the potential to conflict with actual fixes, other developers might work on.

Reply:
As far as I understand, not all changes should fix something. Some changes are made to help other people better understand how it works. In this case, HYPER_SPACE 0xC0400000 won't tell you much, but HYPER_SPACE (PTE_TOP + 1) hints that HYPER_SPACE starts after the PTE area. And I don't understand how this change can create conflict. Even hypothetically. As for the other changes, yes, there are a number of changes that I believe improve the readability of the code, but do not fix anything. If you have specific comments, I suggest discussing them specifically. In fact, this PR is one of the preparatory steps. In general, if this PR creates problems for you, then you can simply close it.

Reply:
Yes, and I totally understand the desire to improve things. But we have made the experience, that "improvements" come at a high cost: 
* We have to review things that do not fix the functionality of the OS, but can easily introduce subtle bugs. This might be useful in some cases, but when it comes to low level kernel functionality, it needs to be seriously justified. We have generally better stuff to do than to review "Make x more consistent" PRs, and we already have plenty of them. Ask Serge, he's a master in creating them.
* All changes that change the code base on a broader scale have the potential to break other developer's code. And we all have partly ready things, that we are working on, and that can fix actual problems, but would suddenly coflict with changes, that were done to "improve" things for them.
* Changing the name of something might make it better to understand what it does, but changing the purpose of an already existing function will make things only worse.

> And I don't understand how this change can create conflict. Even hypothetically. 
Well then make sure the amd64_bringup branch will rebase without conflicts on top of this and everything still works. Good luck! Once you did this, and told me with a straigt face that there was no issue, we will talk again.

Reply:
And don't get me wrong. I don't say your changes are bad or broken. I just want to make sure that we get the changes right and we don't introduce needless changes that don't improve anything. For me this is majorly because I know I will have to deal with it in the x64 branch, i.e. additional work for me witout anything being fixed. I'm willing to accept it, because I don't like preventing improvements and I can deal with it. It's just something you have to bargain about with the core (in this case Mm) deveopers. And that is currentyl mostly me, I guess. Deal with it ;-)

Reply:
I think that apart from
MiProtoPteToPte
MI_MAKE_PROTOTYPE_PTE
MiSubsectionPteToSubsection
MI_MAKE_SUBSECTION_PTE
MI_IS_MAPPED_PTE,
shouldn't have conflicts as 32-bit.
I decided to add them at the last moment.
It makes sense to either divide everything according to architecture, or to make everything universal. I chose the second option.
Unfortunately I have no way to check for 64-bit. My computer is too old)